src/components/SettingsCheckbox/SettingsCheckbox.vue

```diff
@@ -0,0 +1,176 @@
+<!--
+  - @copyright Copyright (c) 2019 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  - @author Greta Doci <gretadoci@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+<docs>
+## One settingscheckbox with a hint
+
+```vue
+<template>
+		<SettingsCheckbox
+			label="some text"
+			hint="some hint"
+			v-model="checkedTarget"/>
+</template>
+```
+</docs>
+
+<template>
+	<input
+		:id="id"
+		type="checkbox"
+		class="checkbox"
+		:checked="value"
+		:disabled="disabled"
+		@keydown.enter.exact.prevent="checkInput"
+		@change="onChange">
+	<label :for="id">{{ label }}</label>
```

Comment:
Not valid, as Eslint says: 'The template root requires exactly one element'

You can put the input inside the label if you want, this is valid and will solve your issue of one root element. 
https://youtu.be/g9Qff0b-lHk